src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -78,11 +78,10 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 
 	// Respond OK as soon as possible, even if we don't have the outbound connection
 	// established yet, to avoid timeouts on the client application
-	success := make(chan bool, 1)
+	success := make(chan bool)
```

Comment:
Oh right whoops -- didn't mean to commit that! Good catch. Yeah I'm also not sure test -race could have caught most of this stuff -- I caught all of these while running.